### ISSUE TITLE
[FR] learning curve visualization

### ISSUE THREAD

**Issue by schmidt-jake** [mlflow/mlflow#7186]
### Willingness to contribute

No. I cannot contribute this feature at this time.

### Proposal Summary

Add a learning curve visualization to the Run Comparison view.

### Motivation

> #### What is the use case for this feature?

A data scientist wants to understand the impact of hyperparameter choices on model convergence.

> #### Why is this use case valuable to support for MLflow users in general?

When comparing runs, charting metrics over time for each run is useful for understanding the impact of hyperparameters and informing model selection.

> #### Why is this use case valuable to support for your project(s) or organization?

See above.

> #### Why is it currently difficult to achieve this use case?

MLFlow (as of version 1.30.0) doesn't implement a learning curve visualization.

### Details

I think this feature would only require touching the frontend code, since the data structure for metrics tracking already exists.

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [X] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [X] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by BenWilson2** [comment]
Hi @schmidt-jake could you expand a bit more on what you're proposing here? 
Is the intention to provide a metric's logging time as an x-axis value with metric values as the y component? If so, how does this fit into hyperparameter optimization run comparison or logging of runs from a distributed platform wherein the relationship between run creation and metric logging might not be relevant? 

If you could draft up a visualization example (in matplotlib or similar framework) that illustrates what you're looking to achieve with this FR it would really help out as well.

**Comment by schmidt-jake** [comment]
@BenWilson2 added a visual to the issue description!

**Comment by BenWilson2** [comment]
@schmidt-jake I'm glad that I guessed what you were talking about. Thank you for confirming. We'll take a look at it internally and discuss. Thank you for the great idea!

**Comment by BenWilson2** [comment]
Out of curiosity, have you tried plotting by `step` (that's recorded with iterative DL training) for the x-axis on the provided plotting iframe within run comparison? Does that fulfill the needs of your FR?

**Comment by schmidt-jake** [comment]
@BenWilson2 tell me if I'm misunderstanding — I tried what you suggested and here's the result when I plot `epoch` (our available equivalent of your `step`):

<img width="1875" alt="Screen Shot 2022-11-07 at 12 48 31 PM" src="https://user-images.githubusercontent.com/13136287/200401276-e6346559-3740-492c-9be3-d2f31f7c385c.png">

The five runs being compared were all very short, only 1 or 2 epochs apiece. And this gets close to the idea, but as you can see, there isn't an appropriate way to identify which datapoints belong to which run.

**Comment by BenWilson2** [comment]
Each of those epoch writes to the tracking server have a discrete timestamp component to them?

**Comment by schmidt-jake** [comment]
> Each of those epoch writes to the tracking server have a discrete timestamp component to them?

I believe so, because they are logged as metrics and each metric is logged with its timestamp.

**Comment by cmantoux** [comment]
Hi there! I just stumbled on this issue while searching for this very feature. Has the situation evolved since last year?

**Comment by B-Step62** [comment]
The run comparison Chart view plots metrics over step for multiple runs as lines, which covers this learning-curve use case. Closing as resolved, feel free to reopen if you need something the existing metric charts do not provide.
